lib/semantics/semantics.cc

```diff
@@ -228,6 +230,45 @@ void DoDumpSymbols(std::ostream &os, const Scope &scope, int indent) {
   --indent;
 }
 
+void Semantics::DumpSymbolsSources(std::ostream &os) {
+  std::list<Symbol> symbols;
+  auto less = [&](Symbol a, Symbol b) {
+    return a.name().begin() < b.name().begin();
+  };
+  std::set<Symbol, decltype(less)> visited(less);
+  GetAllSymbols(os, context_.globalScope(), cooked_, symbols);
+  for (auto symbol : symbols) {
+    auto sourceInfo = cooked_.GetLineAndColumnsFromCooked(symbol.name());
+    if (visited.find(symbol) != visited.end()) {
+      continue;
+    }
+    if (sourceInfo) {
+      os << symbol.name().ToString() << ": " << std::get<0>(*sourceInfo) << ","
+         << std::get<1>(*sourceInfo) << "," << std::get<0>(*sourceInfo) << ","
+         << std::get<2>(*sourceInfo) << "\n";
```

Comment:
Why print element 0 of the tuple (the line number?) twice?

It might be easier to understand the output if the format were:
name: line, col1-col2

Reply:
You're right, that format is easier to understand. line1,col1,line2,col2 format was just easier to parse so I used it, but your suggestion is better.